pkg/odo/cli/service/create.go

```diff
@@ -190,6 +207,31 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 		return nil
 	}
 
+	// we want to find an Operator only if something's passed to the crd flag on CLI
+	if experimental.IsExperimentalModeEnabled() && o.Crd != "" {
+		// make sure that CSV of the specified ServiceType exists
+		csv, err := o.KClient.GetClusterServiceVersion(o.ServiceType)
+		if err != nil {
+			// error only occurs when OperatorHub is not installed.
+			// k8s does't have it installed by default but OCP does
+			return err
+		}
+
+		var almExamples []map[string]interface{}
+		err = json.Unmarshal([]byte(csv.Annotations["alm-examples"]), &almExamples)
+		if err != nil {
+			return errors.Wrap(err, "unable to unmarshal alm-examples")
+		}
+
+		for _, example := range almExamples {
+			if example["kind"].(string) == o.Crd {
+				o.exampleCR = example
+				o.group, o.version = gvFromALMExample(example)
+				o.resource = resourceFromCSV(csv, o.Crd)
+				return nil
```

Comment:
what happens here if none of the almExamples' kind match the `crd` name provided by the user?